src/unix/fs.c

```diff
@@ -404,6 +404,36 @@ static ssize_t uv__fs_readlink(uv_fs_t* req) {
   return 0;
 }
 
+static ssize_t uv__fs_realpath(uv_fs_t* req) {
+  ssize_t len;
+  char* buf;
+
+  len = pathconf(req->path, _PC_PATH_MAX);
+
+  if (len == -1) {
+#if defined(PATH_MAX)
+    len = PATH_MAX;
+#else
+    len = 4096;
+#endif
+  }
+
+  buf = uv__malloc(len + 1);
+
+  if (buf == NULL) {
+    errno = ENOMEM;
+    return -1;
+  }
+
+  if (realpath(req->path, buf) == NULL) {
```

Comment:
True, realpath() does that, but I thought it would be better to allocate using uv__malloc() - for profiling and valgrind. Also, as we freeing that memory using its counterpart uv__free().